### DISCUSSION TITLE
custom dnsmasq entries

### DISCUSSION THREAD

**Discussion by zerog2k** [unifi-utilities/unifi-common#137] [Ideas]
First of thanks for the great extension to UDM.

I see there are some ways to run your own dns (pihole, etc) - but I just wanted a simple way to inject/override some dnsmasq settings into the local UDM's dnsmasq, which is my local resolver for the network. For whatever reasons, they seem to be missing some features, but I have issues with things like static hostname resolution. Not sure what the best way is, but injecting these static records into local dnsmasq config seems to make my network happy.

e.g.
`/mnt/data/on_boot.d/20-dnsmasq_custom.sh`
```
#!/bin/sh

echo "copying custom config then restarting dnsmasq."
/bin/cp -v /mnt/data/on_boot.d/dnsmasq-custom.conf /run/dnsmasq.conf.d/
# SIGHUP does not seem to work for dnsmasq
/bin/kill $(/bin/pidof dnsmasq)
```
and then I can have nice things like:
`/mnt/data/on_boot.d/dnsmasq-custom.conf`
```
# custom dnsmasq config
host-record=spire,10.0.0.8,5
host-record=spire.lan,10.0.0.8,5
host-record=watchtower-pc,10.0.0.4,5
host-record=watchtower-pc.lan,10.0.0.4,5
```

post here in case of any value to others, or if there is perhaps a better way (aside from running various alternate dns servers, which I'm not interested in at this time)

**Comment by nicx** [comment]
@zerog2k Does the solution run reliably and without problems? So the custom config is not overwritten at some point during operation or similar?I have similar plans, see https://github.com/boostchicken/udm-utilities/issues/207

**Reply by zerog2k** [reply]
For my limited purposes (adding some static hosts entry) it has worked reliably.
Not sure about more advanced dnsmasq configs

**Comment by zerog2k** [comment]
fyi, noticed today that turning on "content filtering "for networks (vlans/wifi networks) in new 6.5.x controller will setup some separate custom dnsmasq process instances (aka "dnsfilter") which will each use separate dnsmasq config pointing to external dns server (at cleanbrowsing.org) with no obvious way to inject/override it to use any local entries or servers.
If you do want to use cleanbrowsing.org or opendns familyshield (which I use), just set that up on your internet connection as dns server overrides, and let each dhcp pool on each network use automatic (which should be the internet/wan dns servers you specified), and then leave the content filtering off.
If you are injecting custom dnsmasq directives via this method, then you can add any additional logic/filtering allow/deny listing using this method as well.
Separate discussion, but you may also want to look into limiting 53/udp outbound only to udm/dnsmasq (as to limit anyone's ability to bypass dnsfiltering)
